src/cursor/abstract_cursor.ts

```diff
@@ -671,7 +730,8 @@ export abstract class AbstractCursor<
 
   /** @internal */
   protected abstract _initialize(
-    session: ClientSession | undefined
+    session: ClientSession | undefined,
+    options?: CursorInitializeOptions
```

Comment:
Investigate using this.cursorOptions instead if possible? And this.timeoutContext